<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.33">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2025-08-18">
<meta name="keywords" content="Vestibulum, Nunc ac dignissim, Proin feugiat">

<title>Notes to paperes</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-ea385d0e468b0dd5ea5bf0780b1290d9.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-d24ab4a4f01adf7b6391c077e236a78f.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">


<meta name="citation_title" content="Notes to paperes">
<meta name="citation_keywords" content="Vestibulum,Nunc ac dignissim,Proin feugiat">
<meta name="citation_author" content="Johan Ulstrup">
<meta name="citation_publication_date" content="2025-08-18">
<meta name="citation_cover_date" content="2025-08-18">
<meta name="citation_year" content="2025">
<meta name="citation_online_date" content="2025-08-18">
<meta name="citation_language" content="en">
<meta name="citation_reference" content="citation_title=An introduction to population genetics: Theory and applications;,citation_author=Rasmgb Nielsen;,citation_author=Montgomery Slatkin;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;">
<meta name="citation_reference" content="citation_title=Strong Selective Sweeps on the X Chromosome in the Human-Chimpanzee Ancestor Explain Its Low Divergence;,citation_abstract=The human and chimpanzee X chromosomes are less divergent than expected based on autosomal divergence. We study incomplete lineage sorting patterns between humans, chimpanzees and gorillas to show that this low divergence can be entirely explained by megabase-sized regions comprising one-third of the X chromosome, where polymorphism in the human-chimpanzee ancestral species was severely reduced. We show that background selection can explain at most 10% of this reduction of diversity in the ancestor. Instead, we show that several strong selective sweeps in the ancestral species can explain it. We also report evidence of population specific sweeps in extant humans that overlap the regions of low diversity in the ancestral species. These regions further correspond to chromosomal sections shown to be devoid of Neanderthal introgression into modern humans. This suggests that the same X-linked regions that undergo selective sweeps are among the first to form reproductive barriers between diverging species. We hypothesize that meiotic drive is the underlying mechanism causing these two observations.;,citation_author=Julien Y Dutheil;,citation_author=Kasper Munch;,citation_author=Kiwoong Nam;,citation_author=Thomas Mailund;,citation_author=Mikkel H Schierup;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=8;,citation_doi=10.1371/journal.pgen.1005451;,citation_issn=1553-7390;,citation_pmid=26274919;,citation_volume=11;,citation_journal_title=PLOS Genetics;">
<meta name="citation_reference" content="citation_title=Extraordinary selection on the human X chromosome associated with archaic admixture;,citation_abstract=The X chromosome in non-African humans shows less diversity and less Neanderthal introgression than expected from neutral evolution. Analyzing 162 human male X chromosomes worldwide, we identified fourteen chromosomal regions where nearly identical haplotypes spanning several hundred kilobases are found at high frequencies in non-Africans. Genetic drift alone cannot explain the existence of these haplotypes, which must have been associated with strong positive selection in partial selective sweeps. Moreover, the swept haplotypes are entirely devoid of archaic ancestry as opposed to the non-swept haplotypes in the same genomic regions. The ancient Ust’-Ishim male dated at 45,000 before the present (BP)&nbsp;also carries the swept haplotypes, implying that selection on the haplotypes must have occurred between 45,000 and 55,000 years ago. Finally, we find that the chromosomal positions of sweeps overlap previously reported hotspots of selective&nbsp;sweeps in great ape evolution, suggesting a mechanism of selection unique to X chromosomes.;,citation_author=Laurits Skov;,citation_author=Moisès Coll Macià;,citation_author=Elise Anne Lucotte;,citation_author=Maria Izabel Alvez Cavassim;,citation_author=David Castellano;,citation_author=Mikkel Heide Schierup;,citation_author=Kasper Munch;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_doi=10.1016/j.xgen.2023.100274;,citation_issn=2666-979X;,citation_journal_title=Cell Genomics;">
<meta name="citation_reference" content="citation_title=Extreme selective sweeps independently targeted the X chromosomes of the great apes;,citation_abstract=The unique inheritance pattern of the X chromosome exposes it to natural selection in a way that is different from that of the autosomes, potentially resulting in accelerated evolution. We perform a comparative analysis of X chromosome polymorphism in 10 great ape species, including humans. In most species, we identify striking megabase-wide regions, where nucleotide diversity is less than 20% of the chromosomal average. Such regions are found exclusively on the X chromosome. The regions overlap partially among species, suggesting that the underlying targets are partly shared among species. The regions have higher proportions of singleton SNPs, higher levels of population differentiation, and a higher nonsynonymous-to-synonymous substitution ratio than the rest of the X chromosome. We show that the extent to which diversity is reduced is incompatible with direct selection or the action of background selection and soft selective sweeps alone, and therefore, we suggest that very strong selective sweeps have independently targeted these specific regions in several species. The only genomic feature that we can identify as strongly associated with loss of diversity is the location of testis-expressed ampliconic genes, which also have reduced diversity around them. We hypothesize that these genes may be responsible for selective sweeps in the form of meiotic drive caused by an intragenomic conflict in male meiosis.;,citation_author=Kiwoong Nam;,citation_author=Kasper Munch;,citation_author=Asger Hobolth;,citation_author=Julien Dutheil;,citation_author=Krishna R Veeramah;,citation_author=August E Woerner;,citation_author=Michael F Hammer;,citation_author=Great Project;,citation_author=Thomas Mailund;,citation_author=Mikkel Schierup;,citation_author=Javier Prado-Martinez;,citation_author=Peter H Sudmant;,citation_author=Jeffrey M Kidd;,citation_author=Heng Li;,citation_author=Joanna L Kelley;,citation_author=Belen Lorente-Galdos;,citation_author=Timothy D O’Connor;,citation_author=Gabriel Santpere;,citation_author=Alexander Cagan;,citation_author=Christoph Theunert;,citation_author=Ferran Casals;,citation_author=Hafid Laayouni;,citation_author=Anders E Halager;,citation_author=Maika Malig;,citation_author=Jessica Hernandez-Rodriguez;,citation_author=Irene Hernando-Herraez;,citation_author=Kay Prüfer;,citation_author=Marc Pybus;,citation_author=Laurel Johnstone;,citation_author=Michael Lachmann;,citation_author=Can Alkan;,citation_author=Dorina Twigg;,citation_author=Natalia Petit;,citation_author=Carl Baker;,citation_author=Fereydoun Hormozdiari;,citation_author=Marcos Fernandez-Callejo;,citation_author=Marc Dabad;,citation_author=Michael L Wilson;,citation_author=Laurie Stevison;,citation_author=Cristina Camprubí;,citation_author=Tiago Carvalho;,citation_author=Aurora Ruiz-Herrera;,citation_author=Laura Vives;,citation_author=Marta Mele;,citation_author=Teresa Abello;,citation_author=Ivanela Kondova;,citation_author=Ronald E Bontrop;,citation_author=Anne Pusey;,citation_author=Felix Lankester;,citation_author=John A Kiyang;,citation_author=Richard A Bergl;,citation_author=Elizabeth Lonsdorf;,citation_author=Simon Myers;,citation_author=Mario Ventura;,citation_author=Pascal Gagneux;,citation_author=David Comas;,citation_author=Hans Siegismund;,citation_author=Julie Blanc;,citation_author=Lidia Agueda-Calpena;,citation_author=Marta Gut;,citation_author=Lucinda Fulton;,citation_author=Sarah A Tishkoff;,citation_author=James C Mullikin;,citation_author=Richard K Wilson;,citation_author=Ivo G Gut;,citation_author=Mary Gonder;,citation_author=Oliver A Ryder;,citation_author=Beatrice H Hahn;,citation_author=Arcadi Navarro;,citation_author=Joshua M Akey;,citation_author=Jaume Bertranpetit;,citation_author=David Reich;,citation_author=Mikkel H Schierup;,citation_author=Christina Hvilsom;,citation_author=Aida M Andrés;,citation_author=Jeffrey D Wall;,citation_author=Carlos D Bustamante;,citation_author=Evan E Eichler;,citation_author=Tomas Marques-Bonet;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=20;,citation_doi=10.1073/pnas.1419306112;,citation_issn=0027-8424;,citation_pmid=25941379;,citation_volume=112;,citation_journal_title=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=A fine-scale recombination map of the human–chimpanzee ancestor reveals faster change in humans than in chimpanzees and a strong impact of GC-biased gene conversion;,citation_abstract=Recombination is a major determinant of adaptive and nonadaptive evolution. Understanding how the recombination landscape has evolved in humans is thus key to the interpretation of human genomic evolution. Comparison of fine-scale recombination maps of human and chimpanzee has revealed large changes at fine genomic scales and conservation over large scales. Here we demonstrate how a fine-scale recombination map can be derived for the ancestor of human and chimpanzee, allowing us to study the changes that have occurred in human and chimpanzee since these species diverged. The map is produced from more than one million accurately determined recombination events. We find that this new recombination map is intermediate to the maps of human and chimpanzee but that the recombination landscape has evolved more rapidly in the human lineage than in the chimpanzee lineage. We use the map to show that recombination rate, through the effect of GC-biased gene conversion, is an even stronger determinant of base composition evolution than previously reported.;,citation_author=Kasper Munch;,citation_author=Thomas Mailund;,citation_author=Julien Y Dutheil;,citation_author=Mikkel Schierup;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_issue=3;,citation_doi=10.1101/gr.158469.113;,citation_issn=1088-9051;,citation_pmid=24190946;,citation_volume=24;,citation_journal_title=Genome Research;">
<meta name="citation_reference" content="citation_title=The bonobo genome compared with the chimpanzee and human genomes;,citation_abstract=Sequencing of the bonobo genome shows that more than three per cent of the human genome is more closely related to either the bonobo genome or the chimpanzee genome than those genomes are to each other. The chimpanzee and the bonobo are our species’ two closest living relatives. This paper reports the genome sequence of the bonobo, the last ape to be sequenced. Comparative genomic analyses reveal that more than 3% of the human genome is more closely related to either the bonobo or the chimpanzee genome than these are to each other. The results shed light on the ancestry of the two ape species and might eventually help us to understand the genetic basis of phenotypes that humans share with one or the other ape species. Two African apes are the closest living relatives of humans: the chimpanzee (Pan troglodytes) and the bonobo (Pan paniscus). Although they are similar in many respects, bonobos and chimpanzees differ strikingly in key social and sexual behaviours1,2,3,4, and for some of these traits they show more similarity with humans than with each other. Here we report the sequencing and assembly of the bonobo genome to study its evolutionary relationship with the chimpanzee and human genomes. We find that more than three per cent of the human genome is more closely related to either the bonobo or the chimpanzee genome than these are to each other. These regions allow various aspects of the ancestry of the two ape species to be reconstructed. In addition, many of the regions that overlap genes may eventually help us understand the genetic basis of phenotypes that humans share with one of the two apes to the exclusion of the other.;,citation_author=Kay Prüfer;,citation_author=Kasper Munch;,citation_author=Ines Hellmann;,citation_author=Keiko Akagi;,citation_author=Jason R. Miller;,citation_author=Brian Walenz;,citation_author=Sergey Koren;,citation_author=Granger Sutton;,citation_author=Chinnappa Kodira;,citation_author=Roger Winer;,citation_author=James R. Knight;,citation_author=James C. Mullikin;,citation_author=Stephen J. Meader;,citation_author=Chris P. Ponting;,citation_author=Gerton Lunter;,citation_author=Saneyuki Higashino;,citation_author=Asger Hobolth;,citation_author=Julien Dutheil;,citation_author=Emre Karakoç;,citation_author=Can Alkan;,citation_author=Saba Sajjadian;,citation_author=Claudia Rita Catacchio;,citation_author=Mario Ventura;,citation_author=Tomas Marques-Bonet;,citation_author=Evan E. Eichler;,citation_author=Claudine André;,citation_author=Rebeca Atencia;,citation_author=Lawrence Mugisha;,citation_author=Jörg Junhold;,citation_author=Nick Patterson;,citation_author=Michael Siebauer;,citation_author=Jeffrey M. Good;,citation_author=Anne Fischer;,citation_author=Susan E. Ptak;,citation_author=Michael Lachmann;,citation_author=David E. Symer;,citation_author=Thomas Mailund;,citation_author=Mikkel H. Schierup;,citation_author=Aida M. Andrés;,citation_author=Janet Kelso;,citation_author=Svante Pääbo;,citation_publication_date=2012;,citation_cover_date=2012;,citation_year=2012;,citation_issue=7404;,citation_doi=10.1038/nature11128;,citation_issn=0028-0836;,citation_pmid=22722832;,citation_volume=486;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=DNA language models are powerful predictors of genome-wide variant effects;,citation_author=Mitchell R. Kantorovitz;,citation_author=others;,citation_publication_date=2024-05;,citation_cover_date=2024-05;,citation_year=2024;,citation_fulltext_html_url=https://doi.org/10.1073/pnas.2313724121;,citation_issue=3;,citation_doi=10.1073/pnas.2313724121;,citation_issn=1091-6490;,citation_volume=121;,citation_journal_title=Proceedings of the National Academy of Sciences;,citation_publisher=Proceedings of the National Academy of Sciences;">
<meta name="citation_reference" content="citation_title=Literate programming;,citation_author=Donald E. Knuth;,citation_publication_date=1984-05;,citation_cover_date=1984-05;,citation_year=1984;,citation_fulltext_html_url=https://doi.org/10.1093/comjnl/27.2.97;,citation_issue=2;,citation_doi=10.1093/comjnl/27.2.97;,citation_issn=0010-4620;,citation_volume=27;,citation_journal_title=Comput. J.;,citation_publisher=Oxford University Press, Inc.;">
<meta name="citation_reference" content="citation_title=DNA language models are powerful predictors of genome-wide variant effects;,citation_author=Gonzalo Benegas;,citation_author=Sanjit Singh Batra;,citation_author=Yun S. Song;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_doi=10.1073/pnas.2311219121;,citation_journal_title=PNAS;">
<meta name="citation_reference" content="citation_title=Comprehensive mapping of long-range interactions reveals folding principles of the human genome;,citation_author=Erez Lieberman-Aiden;,citation_author=Nynke L. Van Berkum;,citation_author=Louise Williams;,citation_author=Maxim Imakaev;,citation_author=Tobias Ragoczy;,citation_author=Agnes Telling;,citation_author=Ido Amit;,citation_author=Bryan R. Lajoie;,citation_author=Peter J. Sabo;,citation_author=Michael O. Dorschner;,citation_author=Richard Sandstrom;,citation_author=Bradley Bernstein;,citation_author=M. A. Bender;,citation_author=Mark Groudine;,citation_author=Andreas Gnirke;,citation_author=John Stamatoyannopoulos;,citation_author=Leonid A. Mirny;,citation_author=Eric S. Lander;,citation_author=Job Dekker;,citation_publication_date=2009;,citation_cover_date=2009;,citation_year=2009;,citation_issue=5950;,citation_doi=10.1126/science.1181369;,citation_volume=326;,citation_journal_title=Science;">
<meta name="citation_reference" content="citation_title=Chromatin architecture reorganization during stem cell differentiation;,citation_author=Jesse R. Dixon;,citation_author=Inkyung Jung;,citation_author=Siddarth Selvaraj;,citation_author=Yin Shen;,citation_author=Jessica E. Antosiewicz-Bourget;,citation_author=Ah Young Lee;,citation_author=Zhen Ye;,citation_author=Audrey Kim;,citation_author=Nisha Rajagopal;,citation_author=Wei Xie;,citation_author=Yuanyuan Diao;,citation_author=Chia-Lin Liang;,citation_author=Haifan Zhao;,citation_author=Vikas V. Lobanenkov;,citation_author=Jun S. Song;,citation_author=James A. Thomson;,citation_author=Bing Ren;,citation_publication_date=2015;,citation_cover_date=2015;,citation_year=2015;,citation_issue=7539;,citation_doi=10.1038/nature14222;,citation_volume=518;,citation_journal_title=Nature;">
<meta name="citation_reference" content="citation_title=High-resolution TADs reveal DNA sequences underlying genome organization in flies;,citation_author=Fidel Ramírez;,citation_author=Vivek Bhardwaj;,citation_author=Laura Arrigoni;,citation_author=Kin Chung Lam;,citation_author=Björn A. Grüning;,citation_author=José Villaveces;,citation_author=Bianca Habermann;,citation_author=Asifa Akhtar;,citation_author=Thomas Manke;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_issue=1;,citation_doi=10.1038/s41467-017-02525-w;,citation_volume=9;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Stage-resolved hi-c analyses reveal meiotic chromosome organizational features influencing homolog alignment;,citation_author=Wu Zuo;,citation_author=Guangming Chen;,citation_author=Zhimei Gao;,citation_author=Shuai Li;,citation_author=Yanyan Chen;,citation_author=Chenhui Huang;,citation_author=Juan Chen;,citation_author=Zhengjun Chen;,citation_author=Ming Lei;,citation_author=Qian Bian;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_issue=1;,citation_doi=10.1038/s41467-021-26033-0;,citation_volume=12;,citation_journal_title=Nature Communications;">
<meta name="citation_reference" content="citation_title=Reprogramming of meiotic chromatin architecture during spermatogenesis;,citation_author=Yao Wang;,citation_author=Hanben Wang;,citation_author=Yu Zhang;,citation_author=Zhenhai Du;,citation_author=Wei Si;,citation_author=Suixing Fan;,citation_author=Dongdong Qin;,citation_author=Xiaoyang Chen;,citation_author=Jing Li;,citation_author=Qiang Sun;,citation_author=Wei Xie;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=3;,citation_doi=10.1016/j.molcel.2018.11.019;,citation_volume=73;,citation_journal_title=Molecular Cell;">
<meta name="citation_reference" content="citation_title=Sex ratio and unisexual sterility in hybrid animals;,citation_author=J. B. S. Haldane;,citation_publication_date=1922;,citation_cover_date=1922;,citation_year=1922;,citation_issue=2;,citation_doi=10.1007/BF02983075;,citation_volume=12;,citation_journal_title=Journal of Genetics;">
<meta name="citation_reference" content="citation_title=100 years of haldane’s rule;,citation_author=Finn Cowell;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_issue=2;,citation_doi=10.1111/jeb.14112;,citation_volume=36;,citation_journal_title=Journal of Evolutionary Biology;">
<meta name="citation_reference" content="citation_title=The ecology and evolutionary dynamics of meiotic drive;,citation_author=Anna K. Lindholm;,citation_author=Kelly A. Dyer;,citation_author=Renée C. Firman;,citation_author=Lila Fishman;,citation_author=Wolfgang Forstmeier;,citation_author=Luke Holman;,citation_author=Hanna Johannesson;,citation_author=Ulf Köhler;,citation_author=John F. B. Mitchell;,citation_author=Heidi M. Morgan;,citation_author=Christopher P. Pennell;,citation_author=Nobuo Pomiankowski;,citation_author=Amanda A. Ritchie;,citation_author=Jessica L. Shuker;,citation_author=Tanya R. Snook;,citation_author=Sarah L. Spottiswoode;,citation_author=Russell Bonduriansky;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=4;,citation_doi=10.1016/j.tree.2016.02.001;,citation_volume=31;,citation_journal_title=Trends in Ecology &amp;amp;amp; Evolution;">
</head>

<body class="quarto-light">

<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Notes to paperes</h1>
            <p class="subtitle lead">Some subtitle for my thesis</p>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
            <div class="quarto-title-meta-author">
          <div class="quarto-title-meta-heading">Author</div>
          <div class="quarto-title-meta-heading">Affiliation</div>
          
                <div class="quarto-title-meta-contents">
            <p class="author">Johan Ulstrup </p>
          </div>
                <div class="quarto-title-meta-contents">
                    <p class="affiliation">
                        Aarhus University
                      </p>
                  </div>
                    </div>
        
        <div class="quarto-title-meta">

                      
                <div>
            <div class="quarto-title-meta-heading">Published</div>
            <div class="quarto-title-meta-contents">
              <p class="date">August 18, 2025</p>
            </div>
          </div>
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      <div class="quarto-alternate-formats"><div class="quarto-title-meta-heading">Other Formats</div><div class="quarto-title-meta-contents"><p><a href="index.docx"><i class="bi bi-file-word"></i>MS Word</a></p></div><div class="quarto-title-meta-contents"><p><a href="index.pdf"><i class="bi bi-file-pdf"></i>PDF</a></p></div></div></div>
    </div>


    <div>
      <div class="keywords">
        <div class="block-title">Keywords</div>
        <p>Vestibulum, Nunc ac dignissim, Proin feugiat</p>
      </div>
    </div>

    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#index" id="toc-index" class="nav-link active" data-scroll-target="#index">index</a></li>
  <li><a href="#dna-language-models-are-powerful-predictors-of-genome-wide-variant-effects" id="toc-dna-language-models-are-powerful-predictors-of-genome-wide-variant-effects" class="nav-link" data-scroll-target="#dna-language-models-are-powerful-predictors-of-genome-wide-variant-effects">DNA language models are powerful predictors of genome-wide variant effects</a>
  <ul class="collapse">
  <li><a href="#evolution-on-x" id="toc-evolution-on-x" class="nav-link" data-scroll-target="#evolution-on-x">evolution on X</a></li>
  <li><a href="#chromatin-structure" id="toc-chromatin-structure" class="nav-link" data-scroll-target="#chromatin-structure">Chromatin structure</a></li>
  <li><a href="#ab-configurations" id="toc-ab-configurations" class="nav-link" data-scroll-target="#ab-configurations">A/B configurations</a></li>
  <li><a href="#gpn" id="toc-gpn" class="nav-link" data-scroll-target="#gpn">GPN</a></li>
  <li><a href="#cnn-vs-transformer" id="toc-cnn-vs-transformer" class="nav-link" data-scroll-target="#cnn-vs-transformer">cnn vs transformer</a></li>
  </ul></li>
  <li><a href="#materials-and-methods" id="toc-materials-and-methods" class="nav-link" data-scroll-target="#materials-and-methods">Materials and Methods</a>
  <ul class="collapse">
  <li><a href="#gpn-arcitecture" id="toc-gpn-arcitecture" class="nav-link" data-scroll-target="#gpn-arcitecture">GPN arcitecture</a></li>
  <li><a href="#data-preparation-and-pretraining-workflow" id="toc-data-preparation-and-pretraining-workflow" class="nav-link" data-scroll-target="#data-preparation-and-pretraining-workflow">Data Preparation and Pretraining Workflow</a></li>
  <li><a href="#model-execution-workflow-and-hardware-constraints" id="toc-model-execution-workflow-and-hardware-constraints" class="nav-link" data-scroll-target="#model-execution-workflow-and-hardware-constraints">Model Execution Workflow and Hardware Constraints</a></li>
  <li><a href="#using-hic-to-extract-transtion-areas" id="toc-using-hic-to-extract-transtion-areas" class="nav-link" data-scroll-target="#using-hic-to-extract-transtion-areas">using HIC to extract transtion areas</a></li>
  <li><a href="#distance-between-the-edges" id="toc-distance-between-the-edges" class="nav-link" data-scroll-target="#distance-between-the-edges">distance between the edges</a></li>
  <li><a href="#compare-gc-conent-to-transition-in-macaca" id="toc-compare-gc-conent-to-transition-in-macaca" class="nav-link" data-scroll-target="#compare-gc-conent-to-transition-in-macaca">compare GC conent to transition in macaca</a>
  <ul class="collapse">
  <li><a href="#claude-notes" id="toc-claude-notes" class="nav-link" data-scroll-target="#claude-notes">CLAUDE NOTES</a></li>
  </ul></li>
  <li><a href="#x-chromosome-compartments-and-gc-content-an-unexplored-frontier" id="toc-x-chromosome-compartments-and-gc-content-an-unexplored-frontier" class="nav-link" data-scroll-target="#x-chromosome-compartments-and-gc-content-an-unexplored-frontier">X chromosome compartments and GC content: An unexplored frontier</a></li>
  <li><a href="#cpg-islands" id="toc-cpg-islands" class="nav-link" data-scroll-target="#cpg-islands">CPG islands</a></li>
  </ul></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a>
  <ul class="collapse">
  <li><a href="#notes-for-cpg-islands" id="toc-notes-for-cpg-islands" class="nav-link" data-scroll-target="#notes-for-cpg-islands">notes for cpg islands</a>
  <ul class="collapse">
  <li><a href="#artikler" id="toc-artikler" class="nav-link" data-scroll-target="#artikler">artikler</a></li>
  </ul></li>
  <li><a href="#chr-8" id="toc-chr-8" class="nav-link" data-scroll-target="#chr-8">chr 8</a></li>
  </ul></li>
  <li><a href="#repeat" id="toc-repeat" class="nav-link" data-scroll-target="#repeat">repeat</a></li>
  <li><a href="#notes" id="toc-notes" class="nav-link" data-scroll-target="#notes">notes</a></li>
  <li><a href="#cromatin-structure-and-regulatiopn-of-genes" id="toc-cromatin-structure-and-regulatiopn-of-genes" class="nav-link" data-scroll-target="#cromatin-structure-and-regulatiopn-of-genes">cromatin structure and regulatiopn of genes</a></li>
  <li><a href="#repeats-sorunding-edges" id="toc-repeats-sorunding-edges" class="nav-link" data-scroll-target="#repeats-sorunding-edges">repeats sorunding edges</a>
  <ul class="collapse">
  <li><a href="#three-dimensional-genome-reorganization-during-mouse-spermatogenesis" id="toc-three-dimensional-genome-reorganization-during-mouse-spermatogenesis" class="nav-link" data-scroll-target="#three-dimensional-genome-reorganization-during-mouse-spermatogenesis">Three-dimensional genome reorganization during mouse spermatogenesis</a></li>
  <li><a href="#single-cell-long-read-hi-c-scnanohi-c2" id="toc-single-cell-long-read-hi-c-scnanohi-c2" class="nav-link" data-scroll-target="#single-cell-long-read-hi-c-scnanohi-c2">Single-cell long-read Hi-C, scNanoHi-C2</a></li>
  <li><a href="#d-genome-remodeling-and-homologous-pairing-during-meiotic-prophase" id="toc-d-genome-remodeling-and-homologous-pairing-during-meiotic-prophase" class="nav-link" data-scroll-target="#d-genome-remodeling-and-homologous-pairing-during-meiotic-prophase">3D genome remodeling and homologous pairing during meiotic prophase</a></li>
  <li><a href="#dicer-regulates-repeat-rna-and-chromosome-segregation-maybe" id="toc-dicer-regulates-repeat-rna-and-chromosome-segregation-maybe" class="nav-link" data-scroll-target="#dicer-regulates-repeat-rna-and-chromosome-segregation-maybe">DICER regulates repeat RNA and chromosome segregation (maybe???)</a></li>
  </ul></li>
  <li><a href="#whats-next" id="toc-whats-next" class="nav-link" data-scroll-target="#whats-next">whats next</a></li>
  <li><a href="#genome-wide-coancestry-reveals-details-of-ancient-and-recent-male-driven-reticulation-in-baboons" id="toc-genome-wide-coancestry-reveals-details-of-ancient-and-recent-male-driven-reticulation-in-baboons" class="nav-link" data-scroll-target="#genome-wide-coancestry-reveals-details-of-ancient-and-recent-male-driven-reticulation-in-baboons">Genome-wide coancestry reveals details of ancient and recent male-driven reticulation in baboons</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">



  


<section id="index" class="level1">
<h1>index</h1>
</section>
<section id="dna-language-models-are-powerful-predictors-of-genome-wide-variant-effects" class="level1">
<h1>DNA language models are powerful predictors of genome-wide variant effects</h1>
<p>The idea for this project was to evaluate the Genome-Pretrained Network (GPN) introduced in <span class="citation" data-cites="Kantorovitz2024">(<a href="#ref-Kantorovitz2024" role="doc-biblioref">Benegas, Batra, and Song 2024</a>)</span> and determine whether it could achieve greater accuracy than traditional methods in predicting genome-wide variant effects.</p>
<p>The model is designed as a convolutional neural network (CNN) and takes input sequences with a window size of 512. During training, 15% of the positions within each window are masked to enable the model to learn meaningful representations.</p>
<p>The architecture consists of 25 layers, each structured as follows: a dilated convolution layer, followed by an add-and-norm layer with a skip connection from before the dilated convolution. This is followed by a feedforward layer, another add-and-norm layer, and additional skip connections.</p>
<p>A feedforward layer is a fundamental component of neural networks, where inputs pass through one or more fully connected layers with activation functions, transforming data without looping back. This structure helps the model learn complex representations by applying weighted transformations and non-linearities.</p>
<p>A dilated convolution expands the receptive field of a convolutional layer without increasing the number of parameters or reducing resolution. By spacing kernel elements apart, it captures long-range dependencies in sequences, making it particularly useful in genomic data analysis. When combined, dilated convolutions and feedforward layers enhance a model’s ability to recognize patterns across different scales efficiently.</p>
<p>After passing through the 25 layers, the model produces a contextual embedding with a dimension of 512 (D=512), followed by classification layers. The final layer outputs the probabilities of the four nucleotides at each masked position.</p>
<p>The GPN variant effect prediction score is calculated as the log-likelihood ratio between the alternate (ALT) and reference (REF) alleles. Here, L represents the window length in base pairs, and D denotes the embedding dimension.</p>
<section id="evolution-on-x" class="level2">
<h2 class="anchored" data-anchor-id="evolution-on-x">evolution on X</h2>
<p>The production of gametes in sexually reproducing organisms is a complex, tightly regulated process involving numerous cellular and molecular mechanisms. In males, spermatogenesis—the process by which sperm are formed—progresses through four main stages of differentiation: from germ cells to spermatogonia, then to pachytene spermatocytes, round spermatids, and finally to mature spermatozoa (Wang et al.&nbsp;2019). This process is fundamental to male fertility. Meiosis, a specialized form of cell division, ensures the proper pairing, recombination, and segregation of homologous chromosomes, thereby maintaining genetic integrity and promoting variation. A deep understanding of these molecular events is essential for comprehending inheritance, speciation, population genetics, and the biological causes of infertility.</p>
<p>Sex chromosomes differ from autosomes in several key ways due to their distinct patterns of inheritance and copy number. The Y chromosome is found only in males and exists as a single copy. In contrast, the X chromosome follows a more complex inheritance pattern: females carry two copies, while males carry only one. Consequently, the X chromosome resides two-thirds of the time in females and one-third in males, a distribution that influences how selection acts upon it. In males, the hemizygous nature of the X chromosome means that any mutations or deleterious alleles are fully exposed, without a second copy to mask their effects.</p>
<p>This feature is central to Haldane’s rule (Haldane 1922), which posits that in hybrids between two species, the heterogametic sex (e.g., XY in mammals or ZW in birds) is more likely to be inviable or sterile. While widely observed, the precise causes of this pattern remain a subject of ongoing debate. Several hypotheses have been proposed to explain it. One suggests that incompatibilities involving the Y chromosome can arise if it fails to remain compatible with the X chromosome or autosomes. Another centers on dosage compensation, proposing that hybridization may disrupt the regulatory mechanisms that balance gene expression between sex chromosomes. The dominance hypothesis argues that recessive deleterious alleles may be unmasked in the heterogametic sex, leading to sterility. Additional ideas include the faster evolution of male-biased reproductive genes, which can result in functional mismatches, and the accelerated divergence of X-linked loci compared to autosomes, known as faster-X evolution. Lastly, meiotic drive—genetic conflicts between selfish elements on sex chromosomes—has also been implicated in hybrid sterility.</p>
<p>These hypotheses underscore the complex interplay between sex chromosomes, selection, and hybridization (Cowell 2023). Although Haldane’s rule is consistently observed across diverse taxa—and even across kingdoms—the underlying mechanisms are not universally agreed upon. In many cases, multiple processes may act together (Lindholm et al.&nbsp;2016).</p>
<p>The X chromosome, in particular, appears to be subject to strong and unique selective pressures, especially in primates. This topic is explored in greater depth in the following sections.</p>
</section>
<section id="chromatin-structure" class="level2">
<h2 class="anchored" data-anchor-id="chromatin-structure">Chromatin structure</h2>
<p>Regions under strong selection across primate species—including humans and baboons—are consistently found on the X chromosome, spanning megabase-scale genomic areas. As previously noted, factors such as chromosomal architecture and structural rearrangements may play crucial roles in enabling, regulating, or insulating these regions from selective pressures.</p>
<p>Understanding genome organization and variation is therefore essential for uncovering the mechanisms behind selection. Chromatin has long been recognized as central to gene regulation and cellular function (Lieberman-Aiden et al.&nbsp;2009), in part because the 3D structure of chromosomes brings distant genomic elements into close contact. Disruption of this spatial organization can lead to developmental abnormalities (Dixon et al.&nbsp;2015). Chromatin is organized hierarchically, with multiple levels of structure nested within one another (see Figure 2). At the broadest level, chromatin compartments can span several megabases (Lieberman-Aiden et al.&nbsp;2009), while at finer scales, even 500 base pairs can contribute to subgenic structural organization. Structures such as topologically associating domains (TADs) and chromatin loops, which operate at sub-megabase scales, further help partition regulatory elements and maintain proper genomic function (Ramírez et al.&nbsp;2018; Zuo et al.&nbsp;2021).</p>
<p>kigger på cromatin overgange er der noget der er anderledens i de områder konseveret er der noge specifikke motiver diversiteten</p>
<p>Chromatin, the complex of DNA wrapped around histone proteins and associated with various regulatory factors, plays a central role in the spatial and temporal control of gene expression, as well as in maintaining the structural integrity and stability of the genome. Far from being a static scaffold, chromatin is a highly dynamic entity, subject to extensive remodeling in response to cellular signals, developmental cues, and environmental stimuli. Its organization within the nucleus is hierarchical, ranging from nucleosomes to higher-order domains, and this architecture can differ substantially across cell types, developmental stages, and even between closely related species. One of the key features of chromatin organization is its division into functionally distinct states—broadly classified as euchromatin, which is generally open and transcriptionally active, and heterochromatin, which is compacted and repressive. Transitions between these states, whether developmentally programmed or evolutionarily emergent, are often tightly coupled to shifts in gene regulatory activity and cellular identity.</p>
<p>In this study, we focus on characterizing these chromatin state transitions, with a particular interest in understanding how they are distributed across the genome and whether they exhibit predictable patterns linked to specific sequence features or regulatory elements. We ask whether certain genomic regions are more prone to undergoing structural reorganization and whether such regions harbor conserved sequence motifs, epigenetic signatures, or binding sites for key regulatory proteins. Furthermore, we investigate whether particular chromatin configurations are disproportionately associated with genes involved in critical biological processes or with regions of the genome that show signatures of evolutionary constraint or adaptation.</p>
<p>By examining the diversity of chromatin states and transitions across individuals and lineages, we also aim to uncover broader principles governing chromatin architecture and its functional consequences. These insights may inform our understanding of how chromatin organization evolves, how it contributes to phenotypic diversity, and how its dysregulation may underlie disease states. Ultimately, this work contributes to a growing body of knowledge on the interplay between genome structure, function, and evolution.</p>
</section>
<section id="ab-configurations" class="level2">
<h2 class="anchored" data-anchor-id="ab-configurations">A/B configurations</h2>
<p>One widely studied aspect of chromatin architecture is its partitioning into A and B compartments, as revealed by Hi-C and other genome conformation capture techniques. These compartments represent large-scale topological domains of the genome that differ markedly in their regulatory environment and functional output. A compartments are typically associated with euchromatic regions—open, accessible chromatin that is rich in actively transcribed genes, regulatory elements like enhancers and promoters, and active histone marks such as H3K4me3 and H3K27ac. In contrast, B compartments correspond to heterochromatic, transcriptionally repressive regions characterized by reduced accessibility, fewer active genes, and enrichment of repressive epigenetic marks like H3K9me3 and DNA methylation.</p>
<p>This spatial segregation of the genome into A and B compartments plays a critical role in regulating which genes are expressed in a given cell or context. Genes located in A compartments are more likely to be in close physical proximity to other regulatory elements, transcriptional machinery, and nuclear compartments that promote gene activation, such as nuclear speckles. Conversely, genes embedded in B compartments are more isolated from such transcriptional hubs and may be sequestered near the nuclear lamina, where gene silencing is reinforced.</p>
<p>Importantly, the assignment of a genomic region to either an A or B compartment is not fixed; it can shift in response to developmental signals, environmental stress, or disease states. Such compartment switching is a powerful mechanism of gene regulation. For example, when a gene or regulatory locus moves from a B to an A compartment, it may become transcriptionally activated due to its new, more permissive chromatin environment. Likewise, genes that transition into B compartments often show downregulation or complete silencing. These dynamic shifts allow the cell to orchestrate complex gene expression programs with spatial precision, integrating structural and regulatory layers of genome function.</p>
<p>Moreover, comparative studies across species and populations suggest that A/B compartmentalization is both conserved and evolutionarily flexible. Certain compartments—particularly those associated with developmental and housekeeping genes—are stably maintained across evolutionary time, reflecting strong selective pressure to preserve core regulatory networks. Other regions, especially those involved in species-specific traits or environmental responses, exhibit greater compartmental plasticity. By studying the distribution and dynamics of A/B compartments, especially in non-model organisms and hybrid genomes, we can gain novel insights into how chromatin structure evolves and how it contributes to phenotypic diversity and adaptation.</p>
</section>
<section id="gpn" class="level2">
<h2 class="anchored" data-anchor-id="gpn">GPN</h2>
<p>Recent advances in artificial intelligence have brought powerful tools to genomics, particularly through the development of Genomic Pre-trained Networks (GPNs)—deep learning models trained on large-scale genomic sequence data using architectures inspired by natural language processing. GPNs apply self-supervised learning approaches, analogous to models like BERT or GPT, to the DNA sequence, learning to predict masked or missing bases in context. Through this training, GPNs develop an internal representation of the “language of the genome,” capturing patterns of sequence composition, motif syntax, and long-range dependencies that underlie regulatory function.</p>
<p>A key advantage of GPNs lies in their use of transformer architectures, which differ significantly from traditional models like convolutional neural networks (CNNs). While CNNs are adept at recognizing short, fixed-length patterns—such as transcription factor binding motifs—they struggle with learning dependencies over long genomic distances, such as enhancer-promoter interactions or 3D chromatin looping. Transformers, in contrast, leverage self-attention mechanisms, which allow the model to compare and weigh relationships between all nucleotide positions in a sequence, regardless of distance. This makes them exceptionally well-suited for modeling the complex, multi-scale structure of regulatory DNA.</p>
<p>The utility of GPNs in functional genomics has been demonstrated in tasks such as predicting the effects of noncoding variants, chromatin accessibility, and transcription factor binding. For instance, the model introduced by Benegas and Batra (“DNA language models are powerful predictors of genome-wide variant effects”) showed that transformer-based GPNs outperform prior models in predicting variant impact across a variety of cell types and epigenomic contexts, even without explicit annotations. These models can generate base-resolution functional predictions across the genome, enabling insight into the likely regulatory role of any given sequence segment.</p>
<p>In the context of chromatin structure, GPNs hold particular promise for identifying sequence features that influence or correlate with A/B compartment identity. By learning patterns that distinguish sequences in open (A) versus closed (B) chromatin, GPNs can be used to predict how shifts in compartmentalization may arise from sequence variation. Furthermore, these predictions can be extended across species or populations to assess functional conservation—that is, whether a sequence maintains regulatory potential or compartment association despite genetic divergence.</p>
<p>In this study, we leverage GPN-derived embeddings and predictions to investigate the sequence determinants of chromatin organization. We aim to identify genomic regions where the GPN model predicts strong, conserved regulatory activity, and determine whether these regions also show structural stability in chromatin compartments across individuals or hybrid genomes. This approach allows us to assess whether chromatin transitions are driven primarily by changes in DNA sequence, or whether other regulatory layers (e.g., epigenetic modifications, chromatin remodelers) are required to explain shifts between A and B compartments.</p>
<p>Ultimately, integrating GPN predictions with chromatin conformation data (e.g., from Hi-C) provides a powerful framework for exploring genome function at multiple scales—from base-level regulatory logic to large-scale nuclear organization. These models not only enhance our ability to interpret genomic variation and conservation but also open new avenues for studying the evolution of regulatory architecture in systems such as primates, where hybridization and population structure offer unique opportunities to dissect genome regulation in action.</p>
</section>
<section id="cnn-vs-transformer" class="level2">
<h2 class="anchored" data-anchor-id="cnn-vs-transformer">cnn vs transformer</h2>
<p>Convolutional Neural Networks (CNNs) have been the foundational architecture in computer vision for over a decade. Introduced by LeCun et al.&nbsp;(1998) and popularized with the success of AlexNet (Krizhevsky et al., 2012), CNNs exploit spatial hierarchies in image data through local receptive fields and shared weights. Architectures such as VGGNet (Simonyan &amp; Zisserman, 2015), ResNet (He et al., 2016), and EfficientNet (Tan &amp; Le, 2019) have progressively improved the depth, scalability, and performance of CNNs, enabling state-of-the-art results in tasks such as image classification, object detection, and semantic segmentation.</p>
<p>A simple Convolutional Neural Network (CNN) architecture typically consists of several key types of layers. The first are convolutional layers, which apply learnable filters (also known as kernels) to local regions of the input data. These filters are designed to capture spatial features such as edges, textures, and, as the network deepens, more complex patterns. Following the convolutional operations, activation functions are applied—most commonly the Rectified Linear Unit (ReLU)—to introduce non-linearity into the model, enabling it to learn more complex representations.</p>
<p>Pooling layers are another essential component of CNNs. These layers downsample the feature maps, typically using max pooling or average pooling, which reduces the spatial dimensions of the data while retaining the most salient features. This not only lowers computational requirements but also helps mitigate overfitting by providing a form of translation invariance. Toward the end of the architecture, fully connected layers are often used to perform high-level reasoning. These layers integrate the features extracted by earlier layers and are typically responsible for producing the final classification or regression outputs.</p>
<p>One of the main strengths of CNNs lies in their use of local receptive fields and weight sharing, which makes them highly effective at learning spatial hierarchies and translation-invariant features. Even relatively shallow CNNs can perform well on image-related tasks, particularly when training data is limited.</p>
<p>CNNs offer several advantages. They are parameter-efficient due to filter sharing, which significantly reduces the number of learnable weights compared to fully connected networks. Their strong inductive bias—emphasizing locality and translation invariance—makes them particularly well-suited to visual data. Additionally, CNNs benefit from computational efficiency, as they are compatible with highly optimized GPU-based implementations, and they often require less data than more recent architectures to achieve good performance.</p>
<p>However, CNNs also come with limitations. They tend to focus on local features, which makes capturing long-range dependencies challenging without additional architectural mechanisms. Their receptive field is fixed and limited, so incorporating larger contextual information often requires making the network deeper or using special techniques like dilated convolutions or skip connections. Furthermore, CNN performance can be sensitive to architectural choices such as filter sizes, layer depth, and stride configurations, which often require manual tuning.</p>
<p>These challenges have inspired researchers to explore alternative architectures, most notably Transformer models. Unlike CNNs, Transformers are designed to model global relationships more naturally, though often at the expense of requiring more data and computational resources.</p>
</section>
</section>
<section id="materials-and-methods" class="level1">
<h1>Materials and Methods</h1>
<p>Data Sources: Baboon genome datasets, known hybridization cases, population size data.</p>
<p>Model and Tools: GPN model architecture and usage.</p>
<p>Preprocessing: Alignment, feature extraction, and data cleaning.</p>
<p>Entropy Calculation: Methodology for genome-wide entropy measurement.</p>
<section id="gpn-arcitecture" class="level2">
<h2 class="anchored" data-anchor-id="gpn-arcitecture">GPN arcitecture</h2>
<p>The idea for this project was to evaluate the Genome-Pretrained Network (GPN) introduced in and determine whether it could achieve greater accuracy than traditional methods in predicting genome-wide variant effects.</p>
<p>The model is designed as a convolutional neural network (CNN) and takes input sequences with a window size of 512. During training, 15% of the positions within each window are masked to enable the model to learn meaningful representations.</p>
<p>The architecture consists of 25 layers, each structured as follows: a dilated convolution layer, followed by an add-and-norm layer with a skip connection from before the dilated convolution. This is followed by a feedforward layer, another add-and-norm layer, and additional skip connections.</p>
<p>A feedforward layer is a fundamental component of neural networks, where inputs pass through one or more fully connected layers with activation functions, transforming data without looping back. This structure helps the model learn complex representations by applying weighted transformations and non-linearities.</p>
<p>A dilated convolution expands the receptive field of a convolutional layer without increasing the number of parameters or reducing resolution. By spacing kernel elements apart, it captures long-range dependencies in sequences, making it particularly useful in genomic data analysis. When combined, dilated convolutions and feedforward layers enhance a model’s ability to recognize patterns across different scales efficiently.</p>
<p>After passing through the 25 layers, the model produces a contextual embedding with a dimension of 512 (D=512), followed by classification layers. The final layer outputs the probabilities of the four nucleotides at each masked position.</p>
<p>The GPN variant effect prediction score is calculated as the log-likelihood ratio between the alternate (ALT) and reference (REF) alleles. Here, L represents the window length in base pairs, and D denotes the embedding dimension.</p>
</section>
<section id="data-preparation-and-pretraining-workflow" class="level2">
<h2 class="anchored" data-anchor-id="data-preparation-and-pretraining-workflow">Data Preparation and Pretraining Workflow</h2>
<p>To evaluate the Genome-Pretrained Network (GPN), the initial step involved reproducing the pretraining workflow provided in the original publication. This workflow outlines the procedure for generating a dataset compatible with the model, specifically for masked language modeling over genomic sequences.</p>
<p>Initial attempts to apply the workflow to a baboon genome dataset encountered technical difficulties. These issues were not intrinsic to the baboon data itself but stemmed from the implementation of the workflow. To ensure compatibility with the model and maintain consistency with the original study, the Anatoptis genome—used as the reference in the published work—was selected as a pilot dataset.</p>
<p>After resolving the implementation issues, a dataset suitable for GPN pretraining was successfully generated. The resulting data consisted of masked genomic sequences, formatted according to the model’s input requirements.</p>
</section>
<section id="model-execution-workflow-and-hardware-constraints" class="level2">
<h2 class="anchored" data-anchor-id="model-execution-workflow-and-hardware-constraints">Model Execution Workflow and Hardware Constraints</h2>
<p>Following the successful generation of a pretraining-compatible dataset, the next step involved attempting to run the Genome-Pretrained Network (GPN) using the available inference and training workflows. The documentation provided on the model’s GitHub repository lacked detailed guidance, making it challenging to understand the complete execution process and the rationale behind certain steps.</p>
<p>To proceed, an alternative workflow was identified—one specifically designed for use with the Anatoptis genome. This workflow was adopted as a reference implementation for executing the GPN model.</p>
<p>However, significant hardware limitations emerged during this phase. The original study reported using four NVIDIA A100 GPUs, each equipped with 80 GB of RAM. In contrast, the available system utilized NVIDIA L43R GPUs with 45 GB of RAM, resulting in substantial differences in computational capacity.</p>
<p>As a result, attempts to execute the GPN model encountered frequent out-of-memory (OOM) errors, particularly during forward passes through the deeper layers of the network. These memory constraints posed a major obstacle and significantly hindered reproducibility under the available hardware setup.</p>
<p>Initial efforts to train or fine-tune the Genome-Pretrained Network (GPN) focused on applying the workflow to a baboon genome using a multi-GPU setup. The goal was to replicate the original training environment, which employed four NVIDIA A100 GPUs (80 GB RAM each). However, the available infrastructure utilized NVIDIA L43R GPUs, each with 45 GB of RAM, which introduced several compatibility and performance challenges.</p>
<p>Early testing began with the Anatoptis genome. Attempts to process the entire genome using the Snakemake workflow were unsuccessful due to GPU detection issues. These issues stemmed from the cluster configuration, which prevented the Snakemake environment from correctly interfacing with the available GPU hardware.</p>
<p>After resolving these initial configuration issues, the workflow was executed with GPU support. While this improved performance, the model began encountering memory errors after a few hours of training, even when utilizing two to four GPUs in parallel. In an effort to overcome these constraints, the number of GPUs was incrementally increased—from four to six, and finally to eight—providing an aggregate of 360 GB of GPU RAM.</p>
<p>Despite these efforts, the model continued to encounter out-of-memory (OOM) errors during execution. Even with eight GPUs, the memory demands of the GPN model remained prohibitive under the given cluster configuration. These persistent failures highlighted a significant limitation: the model’s architecture and resource demands exceeded the practical limits of the available hardware.</p>
<p>Ultimately, these constraints prevented successful execution of the GPN model on the tested system.</p>
<p>Given the persistent memory limitations encountered during full-genome processing, an alternative strategy was implemented. Rather than inputting the entire Anatoptis genome, the data was partitioned at the chromosome level. This allowed individual chromosomes to be processed independently, while still allocating the full available GPU resources (eight NVIDIA GPUs with a combined 360 GB of RAM) to each run.</p>
<p>This approach proved successful. When processing a single chromosome at a time, the GPN model was able to execute without encountering memory-related failures. As a result, contextual embeddings were successfully generated for individual chromosomes of the Anatoptis genome.</p>
<p>However, due to time constraints near the conclusion of the project timeline, the extracted embeddings were not subjected to further downstream analysis. Although they were successfully retrieved and stored, no additional modeling, interpretation, or evaluation steps were completed within the project scope.</p>
</section>
<section id="using-hic-to-extract-transtion-areas" class="level2">
<h2 class="anchored" data-anchor-id="using-hic-to-extract-transtion-areas">using HIC to extract transtion areas</h2>
</section>
<section id="distance-between-the-edges" class="level2">
<h2 class="anchored" data-anchor-id="distance-between-the-edges">distance between the edges</h2>
<p>To evaluate the availability of data across genomic distances and mitigate potential bias in downstream analyses, we calculated reverse cumulative counts of compartment edge half-distances. First, edges were identified at compartment transitions between A and B (in either direction). For each consecutive pair of such edges, we measured the genomic distance between their start positions and divided it by two, yielding the half-distance between edges.</p>
<p>We then binned these half-distances and computed the reverse cumulative count for each bin, which represents the number of edges with half-distances greater than or equal to that bin. This approach effectively measures the number of edges capable of contributing data at a given spatial scale. Using this metric allows us to normalize other measurements by the available data density, reducing the risk that apparent trends are driven by varying numbers of contributing edges at different distances rather than true biological effects. Counts were calculated up to 2 Mb to encompass the maximum observed separation.</p>
<div id="fig-edge" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-edge-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/reverse_cumulative_count_v2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Reverse cumulative counts of compartment edge half-distances. Distances were calculated by first identifying sequential transitions between compartments A and B (or B and A). For each pair of transitions, the genomic distance between their start positions was measured and halved, producing the “half-distance” metric. The reverse cumulative count at a given bin shows the number of edges with half-distances greater than or equal to that bin’s center, representing the number of edges that could contribute data to measurements at that spatial scale. Extending the plot to 2 Mb captures the full range of half-distances observed."><img src="illustrations/reverse_cumulative_count_v2.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-edge-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Reverse cumulative counts of compartment edge half-distances. Distances were calculated by first identifying sequential transitions between compartments A and B (or B and A). For each pair of transitions, the genomic distance between their start positions was measured and halved, producing the “half-distance” metric. The reverse cumulative count at a given bin shows the number of edges with half-distances greater than or equal to that bin’s center, representing the number of edges that could contribute data to measurements at that spatial scale. Extending the plot to 2 Mb captures the full range of half-distances observed.
</figcaption>
</figure>
</div>
</section>
<section id="compare-gc-conent-to-transition-in-macaca" class="level2">
<h2 class="anchored" data-anchor-id="compare-gc-conent-to-transition-in-macaca">compare GC conent to transition in macaca</h2>
<p>We compared GC content between A and B chromatin compartments for chromosome 8 and chromosome X across five Macaca cell types: fibroblasts, sperm, round spermatids, pachytene spermatocytes, and spermatogonia. Chromatin compartments were defined from A–B and B–A structural transitions, and GC content was measured separately for each compartment.</p>
<p>For chromosome 8, the A compartment consistently showed higher GC content than the B compartment in all cell types, which aligns with expected genomic organization. In chromosome X, however, the differences between compartments were smaller. In round spermatids, GC content was nearly identical between A and B compartments, while in pachytene spermatocytes, spermatogonia, and sperm, A and B compartments had closely overlapping GC values. Interestingly, in fibroblasts the pattern was reversed, with the B compartment displaying higher GC content than the A compartment see <a href="#fig-GC" class="quarto-xref">Figure&nbsp;2</a>.</p>
<div id="fig-GC" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-GC-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/gc_compartment_analysis_chrX_chr8_combined_20250812_144327.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Figure&nbsp;2: Compartment analysis for chromosomes X and 8 in five Macaca cell types—fibroblasts, sperm, round spermatids, pachytene spermatocytes, and spermatogonia. GC content was remapped to genomic locations where chromatin structure transitions between compartments A and B. The data were separated so that each compartment contains only A or B structure. For each genomic bin, the plot shows the mean GC content, with error bars representing the standard deviation of the mean, adjusted by the effective edge contribution (accounting for how many edges contribute at longer distances).For chromosome 8, the A compartment exhibits higher GC content than the B compartment. In tissues involved in spermatogenesis, chromosome X shows similar GC content in both A and B compartments. In fibroblasts, GC content is higher in the B compartment than in the A compartment."><img src="illustrations/gc_compartment_analysis_chrX_chr8_combined_20250812_144327.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-GC-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Compartment analysis for chromosomes X and 8 in five Macaca cell types—fibroblasts, sperm, round spermatids, pachytene spermatocytes, and spermatogonia. GC content was remapped to genomic locations where chromatin structure transitions between compartments A and B. The data were separated so that each compartment contains only A or B structure. For each genomic bin, the plot shows the mean GC content, with error bars representing the standard deviation of the mean, adjusted by the effective edge contribution (accounting for how many edges contribute at longer distances).For chromosome 8, the A compartment exhibits higher GC content than the B compartment. In tissues involved in spermatogenesis, chromosome X shows similar GC content in both A and B compartments. In fibroblasts, GC content is higher in the B compartment than in the A compartment.
</figcaption>
</figure>
</div>
<section id="claude-notes" class="level3">
<h3 class="anchored" data-anchor-id="claude-notes">CLAUDE NOTES</h3>
</section>
</section>
<section id="x-chromosome-compartments-and-gc-content-an-unexplored-frontier" class="level2">
<h2 class="anchored" data-anchor-id="x-chromosome-compartments-and-gc-content-an-unexplored-frontier">X chromosome compartments and GC content: An unexplored frontier</h2>
<p>The well-established pattern of higher GC content in A compartments versus B compartments on autosomes has <em>not been quantitatively investigated for the X chromosome</em>, despite extensive research on X chromosome compartmentalization. This represents a significant and surprising gap in our understanding of 3D genome organization, particularly given recent discoveries about the unique compartmental architecture of the X chromosome. ## Studies confirm extensive X chromosome compartment research Research on X chromosome compartmentalization has advanced dramatically, particularly with the 2021 discovery that even the inactive X chromosome contains hidden A/B-like compartments. Studies have mapped approximately <em>75 A/B-like compartments on the inactive X</em> and <em>90 compartments on the active X</em> in neural precursor cells, revealing a complex organizational structure previously thought to be absent. The original compartment discovery by Lieberman-Aiden and colleagues in 2009 established that A compartments are gene-rich, GC-rich, and transcriptionally active, while B compartments are gene-poor, AT-rich, and heterochromatic. This principle has been consistently validated across autosomal chromosomes, with studies showing strong correlations (r = 0.80) between compartment identity and GC content. ## The missing comparative analysis Despite extensive searching through major scientific databases and journals, <em>no studies provide quantitative GC content percentages for A versus B compartments specifically on the X chromosome</em>, nor do any offer comparative statistics between X chromosomal and autosomal compartment GC content. A 2017 structural modeling study confirmed that A and B compartments “are found to be associated with GC rich and AT rich sequences, respectively,” but provided no X chromosome-specific data. This gap is particularly striking given that researchers have characterized numerous other aspects of X chromosome compartmentalization, including epigenetic signatures, replication timing, and structural features. The absence of this fundamental comparison likely stems from several factors. X chromosome analysis presents unique technical challenges due to X-inactivation in females and hemizygosity in males, leading many genome-wide studies to exclude sex chromosomes from their analyses. Additionally, the discovery that inactive X chromosomes possess compartmental organization is very recent (2021), suggesting this area requires further investigation. ## X chromosome compartments display unique organizational features The X chromosome exhibits several distinctive compartmental characteristics that could affect GC content patterns differently than autosomes. During X-inactivation, standard A/B compartments undergo remodeling into specialized <em>S1/S2 compartments</em>, where S1 compartments are Xist-rich and correspond to gene-dense regions, while S2 compartments are Xist-poor and correspond to gene-poor regions. These then merge to create the characteristic “compartment-less” superstructure of the inactive X. The inactive X’s A-like compartments differ fundamentally from classical A compartments – they’re enriched in Xist RNA and H3K27me3 repressive marks rather than the active transcription marks (H3K4me3) typical of autosomal A compartments. Recent advances using micro-C and single-cell Hi-C technologies have revealed that the X chromosome maintains distinct <em>bipartite megadomain structures</em> separated by the Dxz4 boundary element, a configuration not seen in autosomes. These megadomains form stepwise during development through cohesin-dependent mechanisms, creating a unique architectural framework that could influence local GC content evolution patterns. Studies in 2024 demonstrated that X-megadomains appear transiently in extraembryonic lineages before establishing the final inactive X structure, representing a developmental trajectory unique to sex chromosomes. ## X-inactivation creates complex relationships with GC content The relationship between X-inactivation and compartment organization adds layers of complexity to potential GC content patterns. <em>Escape genes</em>, which maintain expression from the inactive X, preferentially localize to specific subcompartments and can comprise up to 30% of genes in escape-rich clusters. These genes tend to reside in regions that retain some characteristics of A-type compartments, remaining at the periphery of or outside the Xist RNA domain. SINE elements are enriched near promoters of escape genes, potentially facilitating their resistance to inactivation and creating local environments with distinct sequence composition. The mechanistic pathway of X-inactivation – involving Xist recruitment of HNRNPK, PRC1, and SMCHD1 – drives compartment formation through protein self-association, likely via liquid-liquid phase separation. This process creates spatial segregation between different types of heterochromatin on the inactive X, with S1 compartments enriched in Polycomb marks and S2 compartments enriched in H3K9me2/3-associated proteins. This unique heterochromatin organization could create GC content distributions that differ from the standard active/inactive dichotomy seen in autosomal compartments. ## Comparative analyses reveal broader evolutionary patterns While direct X chromosome compartment GC content data remains absent, comparative genomic studies provide relevant context. Research has demonstrated that <em>X chromosomes show unique recombination suppression patterns</em> that affect GC-biased gene conversion differently than autosomes. The reduced recombination on sex chromosomes influences the efficacy of selection on GC content evolution, potentially creating different equilibrium GC levels in compartments. Analysis across placental mammals revealed that genomic architecture constrains X chromosome evolution, with unique 3D folding patterns conserved across species despite extensive sequence divergence. The X chromosome maintains distinct superloop formations involving escapee loci, which may influence local GC content patterns through altered mutational processes or selection pressures. Studies have shown that active domains on the X localize to territory boundaries where GC content patterns may be influenced by transcriptional accessibility and DNA repair mechanisms that differ from those in autosomal compartments. ## Conclusion The lack of quantitative data comparing GC content between X chromosomal and autosomal A/B compartments represents a significant gap in our understanding of 3D genome organization. While the general principle that A compartments are GC-rich and B compartments are AT-rich is well-established for autosomes, <em>whether this pattern holds equally on the X chromosome remains undemonstrated</em>. Given the X chromosome’s unique architectural features – including specialized compartment types during inactivation, escape gene clustering, and distinctive evolutionary constraints – this comparison would provide crucial insights into the fundamental relationship between chromosome structure, sequence composition, and gene regulation. This unexplored frontier presents an important opportunity for future research to quantitatively characterize how sex chromosome compartmentalization relates to the GC content patterns so well documented in autosomes.</p>
</section>
<section id="cpg-islands" class="level2">
<h2 class="anchored" data-anchor-id="cpg-islands">CPG islands</h2>
<p>CpG islands (CGIs) are typically unmethylated DNA regions that are closely associated with gene promoters and play a crucial role in regulating chromatin accessibility and three-dimensional genome organization. Accumulating evidence demonstrates that changes in CGI methylation, histone modifications, or chromatin-binding proteins can drive or accompany transitions between chromatin compartments, shifting from active A compartments to repressive B compartments or vice versa. Such transitions are frequently observed in developmental processes, cancer progression, X-chromosome inactivation, and transcriptional regulation. In many cases, CGIs act as anchors or boundaries that help define chromatin domains, and their epigenetic state can strongly influence large-scale compartmental reorganization.</p>
<p>Recent research on meiotic chromatin architecture provides insights that support this broader view of CGI function. A study by Wang <em>et al.</em> (2019) examined spermatogenesis in rhesus monkeys and mice using low-input Hi-C and found that canonical topologically associating domains (TADs) dissolve during the pachytene stage of meiosis, even while transcription continues. This observation suggests that TAD integrity is not strictly required for gene expression at this stage. Instead of the standard large-scale A/B compartmentalization, the study identified the emergence of smaller, refined A/B compartments that alternate between transcriptionally active and inactive domains. The formation of these fine-scale compartments was shown to depend on the synaptonemal complex (SC). In SC-defective mutants, canonical TADs reappeared, indicating that the SC suppresses TADs while enabling this unique compartmental structure. Notably, this reorganization of 3D chromatin architecture was conserved between monkeys and mice, suggesting a mammalian-wide regulatory principle (Wang <em>et al.</em>, 2019).</p>
<p>Although CpG islands were not the central focus of this study, the findings are consistent with the hypothesis that CGIs contribute to compartmental organization by functioning as boundary or anchoring elements. Because refined compartment transitions align with transcriptional activity, and CGIs are commonly associated with active transcription start sites, it is likely that they play a role in stabilizing the boundaries between alternating compartments. In this way, CGIs may provide local chromatin insulation while also influencing higher-order genome organization.</p>
<p>we investigatet how the CPG islands where locakedm compared to the compartment transition. this was done by remapping the compartment trasition to CPG islands location on the genome. the outgoing data where then filteret acording to the filter below</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>    A_val <span class="op">=</span> result[</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>        ((result[<span class="st">'comp'</span>] <span class="op">==</span> <span class="st">'A'</span>) <span class="op">&amp;</span> (result[<span class="st">'start'</span>] <span class="op">&lt;</span> <span class="dv">0</span>)) <span class="op">|</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>        ((result[<span class="st">'comp'</span>] <span class="op">==</span> <span class="st">'B'</span>) <span class="op">&amp;</span> (result[<span class="st">'start'</span>] <span class="op">&gt;</span> <span class="dv">0</span>))</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>    ].copy()</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>    B_val <span class="op">=</span> result[</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>        ((result[<span class="st">'comp'</span>] <span class="op">==</span> <span class="st">'A'</span>) <span class="op">&amp;</span> (result[<span class="st">'start'</span>] <span class="op">&gt;</span> <span class="dv">0</span>)) <span class="op">|</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>        ((result[<span class="st">'comp'</span>] <span class="op">==</span> <span class="st">'B'</span>) <span class="op">&amp;</span> (result[<span class="st">'start'</span>] <span class="op">&lt;</span> <span class="dv">0</span>))</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>    ].copy()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In this section, we restricted the analysis to chromosome X (chrX). We separated the data into A and B compartments and computed the absolute distance of each CpG island to the nearest compartment edge. The resulting distributions are shown in Figure <a href="#fig-CPG_normalized_chrx" class="quarto-xref">Figure&nbsp;4</a>. The figure contains five panels—fibroblast, spermatogonia, pachytene spermatocyte, round spermatid, and sperm. In fibroblast, compartments A and B are similar up to ~0.8–1.0 Mb, after which the B compartment shows higher CpG-island counts. In spermatogonia and pachytene spermatocyte, A and B remain similar across distances. In round spermatid, we observe a divergence between compartments with higher counts between ~0.3 and 1.2 Mb. In sperm, A and B are similar up to ~1 Mb; beyond this distance there is little to no additional signal in the B compartment.</p>
<p>Applying the same analysis to chromosome 8 (<a href="#fig-CPG_normalized_chr8" class="quarto-xref">Figure&nbsp;7</a>), fibroblast shows elevated CpG-island counts in compartment A from ~0.6 to 1.4 Mb. In spermatogonia, A and B remain similar. In pachytene spermatocyte and round spermatid, counts are higher in compartment A from ~0 to 1.0 Mb compared with compartment B. The same pattern is observed in sperm, with substantially higher counts in compartment A than in compartment B.</p>
<p>#fig-CPG_normalized_chrx</p>
</section>
</section>
<section id="section" class="level1">
<h1></h1>
<p>#fig-CPG_normalized_chr_ab</p>
<section id="notes-for-cpg-islands" class="level2">
<h2 class="anchored" data-anchor-id="notes-for-cpg-islands">notes for cpg islands</h2>
<p>CpG islands, which are typically unmethylated and associated with gene promoters, play a pivotal role in chromatin accessibility and 3D genome organization. Multiple studies have demonstrated that chromatin compartment transitions (A to B or vice versa) can be driven or accompanied by changes in CpG island methylation, histone modifications, or chromatin-binding proteins. These changes are often observed in development, cancer progression, X-chromosome inactivation, and transcriptional regulation. CGIs frequently serve as anchors or boundaries for chromatin domains and their epigenetic status can correlate with large-scale compartment reorganization, particularly between transcriptionally active A compartments and repressive B compartments.</p>
<section id="artikler" class="level3">
<h3 class="anchored" data-anchor-id="artikler">artikler</h3>
<p>Reprogramming of Meiotic Chromatin Architecture during Spermatogenesis (6) Key Findings Summary This paper investigates how 3D chromatin architecture is reorganized during spermatogenesis—specifically in rhesus monkey and mouse models—using low-input Hi-C (a chromosome conformation capture method).</p>
<p>Major Contributions: Loss and Re-establishment of TADs (Topologically Associating Domains):</p>
<p>During meiosis, particularly in pachytene spermatocytes, most canonical TADs dissolve even though transcription continues. This suggests that TAD structure is not strictly required for active gene expression at this stage.</p>
<p>Emergence of Refined Local A/B Compartments:</p>
<ul>
<li><p>Instead of standard large-scale A/B compartmentalization, fine-grained compartmental structures emerge. These “refined-A/B compartments” alternate between transcriptionally active and inactive domains.</p></li>
<li><p>hese compartments are smaller, and their emergence implies a compartmental organization decoupled from transcription activity.-</p></li>
</ul>
<p>Synaptonemal Complex Dependency:</p>
<ul>
<li><p>These refined compartments require a functional synaptonemal complex (SC). In SC-defective mutants, canonical TADs are restored, suggesting that the SC actively suppresses TADs while enabling fine-scale compartmentalization.</p></li>
<li><p>This is a novel role for the SC in 3D genome reorganization, beyond its known mechanical role in chromosome pairing and crossover formation.</p></li>
</ul>
<p>Conservation Across Species:</p>
<ul>
<li>The unique meiotic chromatin organization found in rhesus monkeys is conserved in mice, suggesting this is a mammalian-wide regulatory principle during gametogenesis.</li>
</ul>
<section id="implication-for-cpg-islands" class="level5">
<h5 class="anchored" data-anchor-id="implication-for-cpg-islands">Implication for CpG Islands:</h5>
<p>While CpG islands (CGIs) are not the primary focus of this paper, the study supports the broader hypothesis that CGIs may serve as boundary or anchoring elements in fine-scale compartment formation during meiosis:</p>
<p>The refined-A/B structure shows domain transitions that correlate with transcription activity, a feature often influenced by CGI presence at promoter regions.</p>
<p>Since CGIs are associated with active transcription start sites, they likely anchor boundaries between alternating compartments, effectively contributing to local chromatin insulation or regulation.</p>
<p><span class="citation" data-cites="article">(<a href="#ref-article" role="doc-biblioref"><strong>article?</strong></a>)</span>{Wang2019, year = {2019}, title = {{Reprogramming of Meiotic Chromatin Architecture during Spermatogenesis}}, author = {Wang, Yu and Wang, Hui and Zhang, Yixiao and Du, Zheng and Si, Wei and Fan, Shuhan and Qin, Dong and Liu, Yuliang and Liu, Yuhan and Xu, Shiyuan and others}, journal = {Molecular Cell}, issn = {1097-4164}, doi = {10.1016/j.molcel.2018.11.019}, abstract = {{Chromatin organization undergoes drastic reconfiguration during gametogenesis. However, the molecular reprogramming of three-dimensional chromatin structure in this process remains poorly understood for mammals, including primates. Here, we examined three-dimensional chromatin architecture during spermatogenesis in rhesus monkey using low-input Hi-C. Interestingly, we found that topologically associating domains (TADs) undergo dissolution and reestablishment in spermatogenesis. Strikingly, pachytene spermatocytes, where synapsis occurs, are strongly depleted for TADs despite their active transcription state but uniquely show highly refined local compartments that alternate between transcribing and non-transcribing regions (refined-A/B). Importantly, such chromatin organization is conserved in mouse, where it remains largely intact upon transcription inhibition. Instead, it is attenuated in mutant spermatocytes, where the synaptonemal complex failed to be established. Intriguingly, this is accompanied by the restoration of TADs, suggesting that the synaptonemal complex may restrict TADs and promote local compartments. Thus, these data revealed extensive reprogramming of higher-order meiotic chromatin architecture during mammalian gametogenesis.}}, volume = {73}, number = {3}, pages = {547–561.e6}, pmid = {30686580}, local-url = {https://www.cell.com/molecular-cell/fulltext/S1097-2765(18)30989-4} }</p>
</section>
<section id="maybe-relevant-articles" class="level5">
<h5 class="anchored" data-anchor-id="maybe-relevant-articles">maybe relevant articles:</h5>
<p>Reconstructing A/B compartments as revealed by Hi-C using long-range correlations in epigenetic data (3)</p>
<p>Smchd1-Dependent and -Independent Pathways Determine Developmental Dynamics of CpG Island Methylation on the Inactive X Chromosome (4)</p>
<div id="fig-CPG_not_normalized" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_not_normalized-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chrx_not_normalized.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;3: chrX"><img src="illustrations/CPG_islands_chrx_not_normalized.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_not_normalized-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: chrX
</figcaption>
</figure>
</div>
<div id="fig-CPG_normalized_chrx" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_normalized_chrx-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chrx_normalized_poisson_CI.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;4: chrX"><img src="illustrations/CPG_islands_chrx_normalized_poisson_CI.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_normalized_chrx-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4: chrX
</figcaption>
</figure>
</div>
<div id="fig-CPG_normalized_ab_chr_x" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_normalized_ab_chr_x-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chrx_normalized_poisson_CI_AB_BA.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;5: chrX"><img src="illustrations/CPG_islands_chrx_normalized_poisson_CI_AB_BA.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_normalized_ab_chr_x-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5: chrX
</figcaption>
</figure>
</div>
</section>
</section>
</section>
<section id="chr-8" class="level2">
<h2 class="anchored" data-anchor-id="chr-8">chr 8</h2>
<div id="fig-CPG_not_normalized" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_not_normalized-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chr8_not_normalized.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;6: chr8"><img src="illustrations/CPG_islands_chr8_not_normalized.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_not_normalized-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: chr8
</figcaption>
</figure>
</div>
<div id="fig-CPG_normalized_chr8" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_normalized_chr8-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chr8_normalized_poisson_CI.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;7: chr8"><img src="illustrations/CPG_islands_chr8_normalized_poisson_CI.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_normalized_chr8-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7: chr8
</figcaption>
</figure>
</div>
<div id="fig-CPG_normalized_chr_ab" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-CPG_normalized_chr_ab-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/CPG_islands_chr8_normalized_poisson_CI_AB_BA.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;8: chr8"><img src="illustrations/CPG_islands_chr8_normalized_poisson_CI_AB_BA.png" class="img-fluid figure-img" style="width:100.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-CPG_normalized_chr_ab-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8: chr8
</figcaption>
</figure>
</div>
</section>
</section>
<section id="repeat" class="level1">
<h1>repeat</h1>
</section>
<section id="notes" class="level1">
<h1>notes</h1>
</section>
<section id="cromatin-structure-and-regulatiopn-of-genes" class="level1">
<h1>cromatin structure and regulatiopn of genes</h1>
<ol start="2" type="1">
<li>Regulation of Gene Expression The chromatin state influences gene accessibility:</li>
</ol>
<p>Open chromatin is associated with active gene transcription.</p>
<p>Closed chromatin represses gene activity. Repetitive elements near regulatory genes can shift chromatin structure, acting as silencers or enhancers depending on the epigenetic landscape (Venkatesh &amp; Workman, 2015).</p>
<p>Venkatesh &amp; Workman (2015) – Histone exchange, chromatin structure and transcription regulation Nature Reviews Molecular Cell Biology Highlights the plasticity of chromatin around repeats and its impact on transcription.</p>
</section>
<section id="repeats-sorunding-edges" class="level1">
<h1>repeats sorunding edges</h1>
<ol type="1">
<li>Genome Stability and Repression of Transposable Elements Repetitive sequences, such as LINEs, SINEs, and satellite DNA, are often silenced via heterochromatin formation. This closed chromatin state is essential to suppress transposable elements, which can otherwise mobilize and disrupt genome integrity (Moazed, 2011).</li>
</ol>
<p>Makova &amp; Hardison (2015) – The effects of chromatin organization on variation in mutation rates Nature Reviews Genetics Discusses how heterochromatin protects repeats but also contributes to mutational heterogeneity.</p>
<ol start="4" type="1">
<li>Development and Cell Differentiation During development, regions with repetitive DNA undergo reconfiguration. The switch from euchromatin to heterochromatin is essential for cell lineage specification, and misregulation can cause developmental disorders (Gelato &amp; Fischle, 2008).</li>
</ol>
<p>Ghosh &amp; Woodcock (2010) – Chromatin higher-order structure and dynamics Cold Spring Harbor Perspectives Explains how chromatin compaction is governed by nucleosome positioning and repeat regions.</p>
<section id="three-dimensional-genome-reorganization-during-mouse-spermatogenesis" class="level3">
<h3 class="anchored" data-anchor-id="three-dimensional-genome-reorganization-during-mouse-spermatogenesis">Three-dimensional genome reorganization during mouse spermatogenesis</h3>
<ul>
<li>Highlights how A/B compartment transitions coincide with major cell state transitions during spermatogenesis.</li>
<li>Found TF motifs and simple repeats enriched in switching regions.</li>
<li>Sheds light on regulatory elements linked to chromatin phase switching.</li>
</ul>
<p>Core Findings This study investigates how the 3D organization of the genome changes across seven sequential stages of mouse spermatogenesis, using Hi-C technology. The work specifically focuses on topologically associating domains (TADs), chromatin loops, and A/B compartments, revealing highly dynamic reorganization events.</p>
<ol type="1">
<li>TAD and Loop Dynamics Present in Type A spermatogonia (early stage).</li>
</ol>
<p>Disappear at pachytene stage (meiosis I prophase).</p>
<p>Reappear in mature spermatozoa.</p>
<p>This pattern indicates that TADs are not static during development and may not be essential for transcription at all stages.</p>
<ol start="2" type="1">
<li>CTCF Binding CTCF remains bound at TAD boundary regions even when TADs are absent (e.g., in pachytene spermatocytes).</li>
</ol>
<p>This suggests CTCF binding is not sufficient by itself to maintain 3D structure.</p>
<ol start="3" type="1">
<li>Enhancers, Promoters, and Active Transcription Despite the absence of TADs at pachytene, enhancers and promoters retain open chromatin.</li>
</ol>
<p>Transcription continues, implying that gene expression can occur independently of higher-order chromatin loops.</p>
<ol start="4" type="1">
<li>A/B Compartment Changes A/B compartments are largely conserved on autosomes throughout spermatogenesis.</li>
</ol>
<p>However, specific A -&gt; B switching events are correlated with changes in gene activity.</p>
<p>The X chromosome loses its A/B compartment structure during stages like pachytene spermatocytes (pacSC), round spermatids (rST), and elongating spermatids (eST).</p>
<p>Repeats and Chromatin Compartment Transitions TF motif enrichment was analyzed in differentially active chromatin regions, which also included repetitive DNA content.</p>
<p>During A/B compartment switching, particularly in regions transitioning from B (closed) to A (open), some simple and low-complexity repeats were enriched, suggesting a role in:</p>
<p>Regulatory rewiring during cell fate transition.</p>
<p>Chromatin accessibility and transcription activation.</p>
<p>The X chromosome, which undergoes massive reorganization and loss of A/B compartment structure during specific stages (pacSC, rST, eST), may exhibit a distinct repeat signature, though this was only briefly hinted at in the paper.</p>
<p><span class="citation" data-cites="article">(<a href="#ref-article" role="doc-biblioref"><strong>article?</strong></a>)</span>{Luo2019, year = {2019}, title = {{Three-dimensional genome reorganization during mouse spermatogenesis}}, author = {Luo, Zhen and Wang, Xin and Wang, Rui and Chen, Jie and Chen, Yan and Xu, Qing and Cao, Jun}, journal = {bioRxiv}, doi = {10.1101/585281}, abstract = {{Dynamic changes in chromatin architecture accompany spermatogenesis. Using Hi-C and ATAC-seq, this study reveals A/B compartment transitions and transcription factor motif enrichments that align with germ cell stage-specific chromatin changes.}}, pages = {}, number = {}, volume = {}, pmid = {}, pmcid = {}, issn = {}, keywords = {}, local-url = {https://www.biorxiv.org/content/10.1101/585281v1} }</p>
</section>
<section id="single-cell-long-read-hi-c-scnanohi-c2" class="level3">
<h3 class="anchored" data-anchor-id="single-cell-long-read-hi-c-scnanohi-c2">Single-cell long-read Hi-C, scNanoHi-C2</h3>
<ul>
<li>Uses single-cell Hi-C to resolve chromatin changes in early germ cells.</li>
<li>Finds TE subfamilies enriched differently in A/B compartments, suggesting a regulatory function of repeat elements.</li>
</ul>
<p>Key Findings Summary Development of scNanoHi-C2</p>
<p>The authors introduce scNanoHi-C2, a novel single-cell long-read Hi-C technique.</p>
<p>It overcomes limitations of standard Hi-C by capturing long-range, multi-contact chromatin interactions in individual cells.</p>
<p>3D Genome Dynamics in Germ Cells</p>
<p>Applied to embryonic germline cells, the technique reveals:</p>
<p>Stage-specific topological reorganization of the genome.</p>
<p>Highly dynamic A/B compartment transitions, even at early embryonic stages.</p>
<p>Repeat Element Profiling</p>
<p>The study shows distinct enrichment patterns of transposable elements (TEs) in A and B compartments.</p>
<p>LINEs and SINEs were differentially represented, correlating with developmental chromatin remodeling.</p>
<p>These repeats may contribute to insulation, compartmental identity, or nuclear architecture.</p>
<p>Chromatin Rewiring during Primordial Germ Cell (PGC) Development</p>
<p>Early embryonic germ cells exhibit weaker TAD boundaries and less stable compartments compared to somatic cells.</p>
<p>As differentiation progresses, compartment strength increases, indicating a progressive maturation of 3D genome organization.</p>
<p>X Chromosome Inactivation Patterns</p>
<p>The paper provides high-resolution views of X chromosome organization, noting changes in compartmental asymmetry and pairing during early germ cell development.</p>
<p><span class="citation" data-cites="article">(<a href="#ref-article" role="doc-biblioref"><strong>article?</strong></a>)</span>{Lu2025, year = {2025}, title = {{Single-cell long-read Hi-C, scNanoHi-C2, details 3D genome reorganization in embryonic-stage germ cells}}, author = {Lu, Jiawei and Li, Wen and Tang, Fuchou}, journal = {Nature Structural &amp; Molecular Biology}, doi = {10.1038/s41594-025-01604-7}, issn = {1545-9985}, abstract = {{This study applies scNanoHi-C2 to reveal high-resolution single-cell 3D chromatin structure changes during germline development. Transposable elements and repeat families were differentially enriched in A/B compartments, impacting developmental regulation.}}, pages = {}, number = {}, volume = {}, pmid = {}, pmcid = {}, keywords = {}, local-url = {https://www.nature.com/articles/s41594-025-01604-7} }</p>
</section>
<section id="d-genome-remodeling-and-homologous-pairing-during-meiotic-prophase" class="level3">
<h3 class="anchored" data-anchor-id="d-genome-remodeling-and-homologous-pairing-during-meiotic-prophase">3D genome remodeling and homologous pairing during meiotic prophase</h3>
<ul>
<li>Simple and LINE repeats enriched in B compartments, involved in meiotic structural integrity.</li>
</ul>
<p>Key Findings Summary This study integrates Hi-C chromatin conformation capture with super-resolution microscopy to uncover how 3D genome architecture is remodeled during meiotic prophase I in both male and female germlines (spermatogenesis and oogenesis) in mice.</p>
<p>Homologous Chromosome Pairing Chromosomes exhibit precise and progressive pairing during prophase I.</p>
<p>The study captures early pairing events in spermatocytes and oocytes, using high-resolution contact maps.</p>
<p>Contact frequency increases significantly in homologous regions during zygotene and pachytene stages.</p>
<ol start="2" type="1">
<li>Chromatin Compartment Remodeling A/B compartments are weakened or restructured during prophase.</li>
</ol>
<p>Specific B compartments, enriched in LINE elements and heterochromatin, show elevated pairing dynamics.</p>
<p>Suggests B compartment flexibility supports pairing more than A.</p>
<ol start="3" type="1">
<li>Role of Repeat Elements Repeat-rich regions, especially LINEs and LTRs, are preferentially localized to B compartments and correlate with stronger interhomolog interactions.</li>
</ol>
<p>Repeats may play a structural role in mediating nuclear architecture during meiosis.</p>
<ol start="4" type="1">
<li>Sex Differences in Chromosome Pairing The authors find distinct dynamics between male and female germ cells, including:</li>
</ol>
<p>Differences in compartment switching speed.</p>
<p>Asynchronous timing in homolog pairing progression.</p>
<p>Oocytes retain more stable open chromatin than spermatocytes during mid-prophase.</p>
<p><span class="citation" data-cites="article">(<a href="#ref-article" role="doc-biblioref"><strong>article?</strong></a>)</span>{He2023, year = {2023}, title = {{3D genome remodeling and homologous pairing during meiotic prophase of mouse oogenesis and spermatogenesis}}, author = {He, Jianrong and Yan, Ao and Chen, Bohan and Huang, Jin and Kee, Kehkooi}, journal = {Developmental Cell}, doi = {10.1016/j.devcel.2023.07.004}, issn = {1534-5807}, abstract = {{By applying Hi-C and super-resolution microscopy, this study tracks chromatin compartment transitions and homolog pairing in meiosis. Repeat-enriched B compartments show increased pairing dynamics and regulate meiotic architecture.}}, pages = {}, number = {}, volume = {}, pmid = {}, pmcid = {}, keywords = {}, local-url = {https://www.cell.com/developmental-cell/abstract/S1534-5807(23)00553-1} }</p>
</section>
<section id="dicer-regulates-repeat-rna-and-chromosome-segregation-maybe" class="level3">
<h3 class="anchored" data-anchor-id="dicer-regulates-repeat-rna-and-chromosome-segregation-maybe">DICER regulates repeat RNA and chromosome segregation (maybe???)</h3>
<ul>
<li>Satellite repeats involved in chromatin architecture, A/B compartment boundary insulation via RNA-based silencing pathways.</li>
</ul>
<p><span class="citation" data-cites="article">(<a href="#ref-article" role="doc-biblioref"><strong>article?</strong></a>)</span>{He2023, year = {2023}, title = {{3D genome remodeling and homologous pairing during meiotic prophase of mouse oogenesis and spermatogenesis}}, author = {He, Jianrong and Yan, Ao and Chen, Bohan and Huang, Jin and Kee, Kehkooi}, journal = {Developmental Cell}, doi = {10.1016/j.devcel.2023.07.004}, issn = {1534-5807}, abstract = {{By applying Hi-C and super-resolution microscopy, this study tracks chromatin compartment transitions and homolog pairing in meiosis. Repeat-enriched B compartments show increased pairing dynamics and regulate meiotic architecture.}}, pages = {}, number = {}, volume = {}, pmid = {}, pmcid = {}, keywords = {}, local-url = {https://www.cell.com/developmental-cell/abstract/S1534-5807(23)00553-1} }</p>
</section>
</section>
<section id="whats-next" class="level1">
<h1>whats next</h1>
<p>can we show this 3. Mutation Rate Variation and Evolution Repetitive DNA in closed chromatin accumulates mutations at different rates compared to open regions, contributing to genomic variation and evolution (Makova &amp; Hardison, 2015).</p>
<div id="fig-fig_1_kantorovitz" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-fig_1_kantorovitz-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="illustrations/fig_1_kantorovitz.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;9: "><img src="illustrations/fig_1_kantorovitz.png" class="img-fluid figure-img" style="width:50.0%"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-fig_1_kantorovitz-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9
</figcaption>
</figure>
</div>
</section>
<section id="genome-wide-coancestry-reveals-details-of-ancient-and-recent-male-driven-reticulation-in-baboons" class="level1">
<h1>Genome-wide coancestry reveals details of ancient and recent male-driven reticulation in baboons</h1>
<p><span class="citation" data-cites="knuth84">(<a href="#ref-knuth84" role="doc-biblioref">Knuth 1984</a>)</span></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Here is example python code</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"Hello world"</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="bu">print</span>(<span class="st">"i need to make a update"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Here is a reference <span class="citation" data-cites="NielsenSlatkin2016">(<a href="#ref-NielsenSlatkin2016" role="doc-biblioref">Nielsen and Slatkin 2016</a>)</span></p>
</section>
<section id="references" class="level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Kantorovitz2024" class="csl-entry" role="listitem">
Benegas, Gonzalo, Sanjit Singh Batra, and Yun S. Song. 2024. <span>“DNA Language Models Are Powerful Predictors of Genome-Wide Variant Effects.”</span> <em>PNAS</em>. <a href="https://doi.org/10.1073/pnas.2311219121">https://doi.org/10.1073/pnas.2311219121</a>.
</div>
<div id="ref-knuth84" class="csl-entry" role="listitem">
Knuth, Donald E. 1984. <span>“Literate Programming.”</span> <em>Comput. J.</em> 27 (2): 97–111. <a href="https://doi.org/10.1093/comjnl/27.2.97">https://doi.org/10.1093/comjnl/27.2.97</a>.
</div>
<div id="ref-NielsenSlatkin2016" class="csl-entry" role="listitem">
Nielsen, Rasmgb, and Montgomery Slatkin. 2016. <em>An Introduction to Population Genetics: Theory and Applications</em>.
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->
<script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>